### PR TITLE
Introduced `Storefront` to abstract SK1 `SKStorefront` and SK2 `StoreKit.Storefront`

### PR DESCRIPTION
**Pull request by NachoSoto**
To simplify #1499.

### CODE REVIEW

File: Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift [hunk 1]
```diff
@@ -94,7 +94,7 @@ private extension ProductsFetcherSK2 {
         // Note: not using `Storefront.updates` because it's not currently possible
         // to avoid race conditions. Updates never arrive on time for tests to detect
         // changes and clear the cache.
-        get async { await Storefront.current?.id }
+        get async { await StoreKit.Storefront.current?.id }
```
Comment (NachoSoto) [inline Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift:97]: Thoughts on adding an extension to `Storefront` like this?
```swift
public extension Storefront {

    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, macCatalyst 13.1, *)
    private static var currentStorefrontType: StorefrontType? {
        get async {
            if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) {
                let sk2Storefront = await StoreKit.Storefront.current
                return sk2Storefront.map(SK2Storefront.init)
            } else {
                return SKPaymentQueue.default().storefront.map(SK1Storefront.init)
            }
        }
    }

    /// The current App Store storefront for the device.
    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, macCatalyst 13.1, *)
    static var current: Storefront? {
        get async {
            return await self.currentStorefrontType.map(Storefront.from(storefront: ))
        }
    }

}
```

Then changing this to `Storefront.current` and users can use it too while supporting SK1.

Comment (taquitos) [reply Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift:97]: Makes sense to me 😄

File: Sources/Purchasing/StoreKitAbstractions/Storefront.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    @objc public var countryCode: String { self.storefront.countryCode }
+    @objc public var identifier: String { self.storefront.identifier }
+
+    // swiftlint:enable missing_docs
+
+    // MARK: -
+
+    /// Creates an instance from any `StorefrontType`.
+    /// If `discount` is already a wrapped `Storefront` then this returns it instead.
```
Comment (taquitos) [inline Sources/Purchasing/StoreKitAbstractions/Storefront.swift:45]: `discount` copy pasta 😄

Comment (NachoSoto) [reply Sources/Purchasing/StoreKitAbstractions/Storefront.swift:45]: Oops you got me.

File: Tests/APITesters/ObjCAPITester/ObjCAPITester/main.m [hunk 3]
```diff
@@ -19,6 +19,7 @@
 #import "RCRefundRequestStatusAPI.h"
 #import "RCStoreProductAPI.h"
 #import "RCStoreProductDiscountAPI.h"
+#import "RCStorefrontAPI.h"
```
Comment (taquitos) [inline Tests/APITesters/ObjCAPITester/ObjCAPITester/main.m:22]: alphabetical order

**Review by taquitos — APPROVED** [review]
🐐 a couple tiny comments, but otherwise looks great, thanks!

File: Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift (hunk 1 above)
Comment (NachoSoto) [reply Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift:97]: I'll add it in another PR.

File: Tests/APITesters/ObjCAPITester/ObjCAPITester/main.m (hunk 3 above)
Comment (NachoSoto) [reply Tests/APITesters/ObjCAPITester/ObjCAPITester/main.m]: Oops I did it right below and messed it up here.

**Comment by NachoSoto** [comment]
Changed.

File: Sources/Purchasing/StoreKitAbstractions/Storefront.swift (hunk 2 above)
Comment (taquitos) [reply Sources/Purchasing/StoreKitAbstractions/Storefront.swift]: 😆

File: Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift (hunk 1 above)
Comment (NachoSoto) [reply Sources/Purchasing/StoreKit2/ProductsFetcherSK2.swift:97]: Done in #1502

**Pull request merged**